src/settings-ui/Microsoft.PowerToys.Settings.UI.Library/ViewModels/PowerLauncherViewModel.cs

```diff
@@ -193,113 +188,72 @@ public int MaximumNumberOfResults
             }
         }
 
-        public bool IsDarkThemeRadioButtonChecked
-        {
-            get
-            {
-                return _isDarkThemeRadioButtonChecked;
-            }
-
-            set
-            {
-                if (value == true)
-                {
-                    settings.Properties.Theme = Theme.Dark;
-                    _isDarkThemeRadioButtonChecked = value;
-
-                    UpdateSettings();
-                }
-            }
-        }
-
-        public bool IsLightThemeRadioButtonChecked
-        {
-            get
-            {
-                return _isLightThemeRadioButtonChecked;
-            }
-
-            set
-            {
-                if (value == true)
-                {
-                    settings.Properties.Theme = Theme.Light;
-                    _isDarkThemeRadioButtonChecked = value;
-
-                    UpdateSettings();
-                }
-            }
-        }
-
-        public bool IsSystemThemeRadioButtonChecked
+        public int ThemeIndex
         {
             get
             {
-                return _isSystemThemeRadioButtonChecked;
+                return _themeIndex;
             }
 
             set
             {
-                if (value == true)
+                if (_themeIndex != value)
                 {
-                    settings.Properties.Theme = Theme.System;
-                    _isDarkThemeRadioButtonChecked = value;
-
-                    UpdateSettings();
+                    if (value == 0)
+                    {
```

Comment:
They way this is written is very clear, but also extremely redundant
I would use a temporary array or something, and use the "index" as a real index.
``` cs
array = ["Dark","Light","System"]; // temporary
settings.Properties.Theme = Theme.array[value];
UpdateSettings();
```
(Same for monitor position et cetera)

Reply:
Yes, you are correct. Replaced it with a switch case.